UI/CwCheatScreen.cpp

```diff
@@ -40,10 +41,69 @@ static bool enableAll = false;
 static std::vector<std::string> cheatList;
 static CWCheatEngine *cheatEngine2;
 static std::deque<bool> bEnableCheat;
+std::string gamePath_;
+
+
+CwCheatScreen::CwCheatScreen(std::string gamePath)
+	: UIDialogScreenWithBackground() {
+	gamePath_ = gamePath;
+}
+
+inline void trim2(std::string& str) {
+	size_t pos = str.find_last_not_of(' ');
+	if (pos != std::string::npos) {
+		str.erase(pos + 1);
+		pos = str.find_first_not_of(' ');
+		if (pos != std::string::npos) str.erase(0, pos);
+	}
+	else str.erase(str.begin(), str.end());
+}
 
 void CwCheatScreen::CreateCodeList() {
-	cheatEngine2 = new CWCheatEngine();
-	cheatList = cheatEngine2->GetCodesList();
+	std::string line;
+	std::vector<std::string> codesList;  // Read from INI here
+
+	GameInfo *info = g_gameInfoCache->GetInfo(NULL, gamePath_, 0);
+	std::string discID;
+	if (info && info->paramSFOLoaded) {
+		
+		discID = info->paramSFO.GetValueString("DISC_ID");
+	}
+
+	activeCheatFile = GetSysDirectory(DIRECTORY_CHEATS) + discID + ".ini";
+	File::CreateFullPath(GetSysDirectory(DIRECTORY_CHEATS));
+
+	if (!File::Exists(activeCheatFile)) {
```

Comment:
Hmm, instead of abusing a global `activeCheatFile`, can we make `GetCodesList` take an optional path argument?  I'm thinking:

```c++
class CWCheatEngine ... {
    std::vector<std::string> GetCodesList(std::string file = "");
};

//

std::vector<std::string> CWCheatEngine::GetCodesList(std::string file) {
    if (file.empty()) {
        file = activeCheatFile;
    }
...
```

That way we don't need to duplicate this code, I think?

-[Unknown]